utils/generic.py

```diff
@@ -308,3 +308,184 @@ def annotation_type_in_vcf_info(t: Any) -> bool:
+    :return:
+    """
+
+    def phase_trio_genotypes(proband_gt: hl.expr.CallExpression,
+                             father_gt: hl.expr.CallExpression,
+                             mother_gt: hl.expr.CallExpression,
+                             n_alleles: int) -> hl.expr.ArrayExpression:
+        """
+        Given a trio genotype and the number of alleles at the site, phases the genotypes based on allele transmission in the trio.
+
+        :param CallExpression proband_gt: proband genotype
+        :param CallExpression father_gt: father genotype
+        :param CallExpression mother_gt: mother genotype
+        :param int n_alleles: Number of alleles at the site
+        :return: Array of genotypes (proband, father, mother)
+        :rtype: ArrayExpression
+        """
+
+        def gt_to_vectors(gt: hl.expr.CallExpression, n_alleles: int) -> hl.expr.ArrayExpression:
+            """
+            Get the set of all allele-vectors in a genotype.
+            It is returned as an ordered array where the first vector corresponds to the first allele,
+            and the second vector (only present if het) the second allele.
+
+            :param CallExpression gt: genotype
+            :param int n_alleles: number of alleles at the site
+            :return:
+            """
+            return hl.cond(
+                gt.is_het(),
+                hl.array([
+                    allele_to_vector(gt, 0, n_alleles),
+                    allele_to_vector(gt, 1, n_alleles)
+                ]),
+                hl.array([allele_to_vector(gt, 0, n_alleles)])
+            )
+
+        def allele_to_vector(gt: hl.expr.CallExpression, allele_index: int, n_alleles: int) -> hl.expr.ArrayExpression:
+            """
+            Converts an allele from a genotype into a one-hot-encoded vector of n_alleles length.
+            E.g. gt = 0/1
+            => allele_to_vector(gt, 0, 3) => [1, 0, 0]
+            => allele_to_vector(gt, 1, 3) => [0, 1, 0]
+
+            :param CallExpression gt: Genotype
+            :param int allele_index: Allele index
+            :param int n_alleles: Number of alleles at the site
+            :return: One-hot-encoded genotype-allele array
+            :rtype: ArrayExpression
+            """
+            return hl.range(0,n_alleles).map(lambda i: hl.cond(gt[allele_index]==i, 1, 0))
+
+        def get_phased_parent_gt(gt: hl.expr.CallExpression, transmitted_allele_index: int):
+            """
+            Given a genotype and which allele was transmitted to the offspring, returns the parent phased genotype.
+
+            :param CallExpression gt: Parent genotype
+            :param int transmitted_allele_index: index of transmitted allele (0 or 1)
+            :return: Phased parent genotype
+            :rtype: CallExpression
+            """
+            return hl.call(
+                gt[transmitted_allele_index],
+                hl.cond(
```

Comment:
I think you can probably `gt[hl.int(transmitted_allele_index == 0)]`

Reply:
good call!